### ISSUE TITLE
IntelliSense lists unfitting generic extension methods

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#38382]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/705931/intellisense-lists-unfitting-generic-extension-met.html)._

---
Hence the following code:

```cs
using System;
using System.IO;

namespace ExampleLib
{
    public class Class1
    {
        public void Test()
        {
            object x;
            x. // use IntelliSense here
        }
    }

public static class StreamExtensions
    {
        public static T Read&lt;T&gt;(this T self, out int result) where T : Stream
        {
            result = 1234; // sample code
            return self;
        }
    }
}
```

I defined an extension method using a generic type parameter for the extended class, but it is limited to `Stream` instances due to the type constraint on `T`. This was done to return an extended, typed instance to allow chain calling such methods.

However, when typing `x.` at the marked position in `Class1::Test`, IntelliSense also lists the `Stream` only extension method `Read&lt;&gt;` even though `object` is not guaranteed to be of type `Stream`. The full list shown by IntelliSense then is
- Equals
- GetHashCode
- GetType
- Read&lt;&gt;
- ToString

Is this expected behavior? Of course, theoretically `x` could be passed by calling the method &quot;normally&quot; without syntactic extension method sugar. However, it turns generic extension methods into IntelliSense-cluttering bad guys.

Can this behavior be made optional?

---
#### Visual Studio Feedback System on 8/22/2019, 09:36 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by RikkiGibson** [comment]
Looks like a duplicate of #37780

**Comment by RikkiGibson** [comment]
The fix will go out in VS 16.3
